### ISSUE TITLE
Version 6.3.2 no Login to Miele API after editing password

### ISSUE THREAD

**Issue by rabbit6984** [Grizzelbee/ioBroker.mielecloudservice#269]
Hi,

when i restart the adapter i get the following messages (For Information private Data in [] are removed ).

I reinstalled the adapter without success
i generate a new client id and key no success

so i posted here ;-) sorry for bad english.

mielecloudservice.0 | 2022-07-11 15:27:33.735 | info | Terminated (ADAPTER_REQUESTED_TERMINATION): Terminating adapter due to inability to authenticate.
-- | -- | -- | --
mielecloudservice.0 | 2022-07-11 15:27:33.734 | info | Login attempt wasn't successful. Connection retry in 31.569 Seconds.
mielecloudservice.0 | 2022-07-11 15:27:33.733 | error | IMPORTANT!! Mask/Delete your credentials when posting your log online!
mielecloudservice.0 | 2022-07-11 15:27:33.733 | warn | options country: [de-DE]
mielecloudservice.0 | 2022-07-11 15:27:33.732 | warn | options Client_Secret: []
mielecloudservice.0 | 2022-07-11 15:27:33.732 | warn | options Client_ID: []
mielecloudservice.0 | 2022-07-11 15:27:33.731 | warn | options Miele_Password: []
mielecloudservice.0 | 2022-07-11 15:27:33.731 | warn | options Miele_account: []
mielecloudservice.0 | 2022-07-11 15:27:33.730 | warn | Credentials used for login:
mielecloudservice.0 | 2022-07-11 15:27:33.730 | error | Error: Unable to authenticate user! Your credentials seem to be invalid. Please double check and fix them.
mielecloudservice.0 | 2022-07-11 15:27:33.512 | info | Login attempt #1 @Miele-API
mielecloudservice.0 | 2022-07-11 15:27:33.469 | info | starting. Version 6.3.2 in /opt/iobroker/node_modules/iobroker.mielecloudservice, node: v14.19.3, js-controller: 4.0.23
mielecloudservice.0 | 2022-07-11 15:27:33.215 | info | Plugin sentry Sentry Plugin disabled for this process because sending of statistic data is disabled for the system

**Comment by Grizzelbee** [comment]
Hi @rabbit6984 

going from what I see here your creadentials seem to be invalid. Currently  that's all I can say.

But here are some possible/common mistakes: 
* It's possible that you have leading/trailing blanks in one or more of the fields. Please check this. Use copy/paste!
* Maybe you have mixed up client-secret/client-ID?
* Maybe you have mixed up client-secret/App-password?
* Maybe you have mixed up client-ID/App-ID?
* Test your credentials on the Miele developer homepage: https://www.miele.com/developer/swagger-ui/swagger.html (The green Authorise Button)

Please double check everything.

But ... you wrote: 
> when i restart the adapter i get the following messages 

Only on restarts or aren't you able to login at all?

**Comment by rabbit6984** [comment]
HI  @Grizzelbee 

so i tested it again.
Username and Passwort are correct.
I can login in the smarthpone app with same credentials

I see in the log the plain text from client secret and client passwort are not correct in the [ ]. 
The Username and API Key are in plain text correct between the [ ] 

Would i start the adapter i get the error messages from first post. Thann the adapter are red. I need to restart the adatpter to get the error from post one again.

Thanks

**Comment by Grizzelbee** [comment]
Okay. I can confirm this bug. 

Happens on my side too after playing around with the credentials.
I have to digg deeper and search for the reason.
Please be patient I'll keep you updated while searching for a solution.

**Comment by Grizzelbee** [comment]
It seems that something has changed inside the core of iBroker. Automatic password decryption works now for some reason. So I removed the manual decryption which caused a duplicate decryption and with this the error.

**Comment by Grizzelbee** [comment]
Note: Please provide some more system information:
* node: v14.19.3, 
* js-controller: 4.0.23
* mieleCloudService: Version 6.3.2
* OS: 
* Admin: vx.x.x

**Comment by rabbit6984** [comment]
Hi @Grizzelbee 
Problem ist fix thanks

node 16.60
js-controler 4.0.23
miele cloud service 6.3.3
OS: Raspberry
Admin 6.2.0

thanks

**Comment by MartiMan79** [comment]
@rabbit6984 
> Hi @Grizzelbee Problem ist fix thanks
> 
> node 16.60 js-controler 4.0.23 miele cloud service 6.3.3 OS: Raspberry Admin 6.2.0
> 
> thanks

Hi, what fixed this for you?
I'm facing the same issue...

@Grizzelbee 
Ran the adapter successfully in the past but deleted the instance only (adapter still installed) as not needed.
Now I want Siri to inform me regarding time left for several appliances.
Reinstalled adapter, created new instances (*.1 in case of old config data not being overwritten), registered a new API token and even updated my app credentials but no success.
Version 6.5.4


```
2023-12-26 11:41:32.152 - info: host.rpi4 instance system.adapter.mielecloudservice.1 started with pid 493619
--
2023-12-26 11:41:33.668 - debug: mielecloudservice.1 (493619) Redis Objects: Use Redis connection: 127.0.0.1:9001
2023-12-26 11:41:33.707 - debug: mielecloudservice.1 (493619) Objects client ready ... initialize now
2023-12-26 11:41:33.710 - debug: mielecloudservice.1 (493619) Objects create System PubSub Client
2023-12-26 11:41:33.711 - debug: mielecloudservice.1 (493619) Objects create User PubSub Client
2023-12-26 11:41:33.742 - debug: mielecloudservice.1 (493619) Objects client initialize lua scripts
2023-12-26 11:41:33.750 - debug: mielecloudservice.1 (493619) Objects connected to redis: 127.0.0.1:9001
2023-12-26 11:41:33.754 - silly: mielecloudservice.1 (493619) redis psubscribe cfg.o.system.user.*
2023-12-26 11:41:33.789 - silly: mielecloudservice.1 (493619) redis psubscribe cfg.o.enum.*
2023-12-26 11:41:33.792 - silly: mielecloudservice.1 (493619) objectDB connected
2023-12-26 11:41:33.795 - debug: mielecloudservice.1 (493619) Redis States: Use Redis connection: 127.0.0.1:9000
2023-12-26 11:41:33.806 - debug: mielecloudservice.1 (493619) States create System PubSub Client
2023-12-26 11:41:33.808 - debug: mielecloudservice.1 (493619) States create User PubSub Client
2023-12-26 11:41:33.868 - debug: mielecloudservice.1 (493619) States connected to redis: 127.0.0.1:9000
2023-12-26 11:41:33.869 - silly: mielecloudservice.1 (493619) statesDB connected
2023-12-26 11:41:33.912 - debug: mielecloudservice.1 (493619) Plugin sentry Do not initialize Plugin (enabled=false)
2023-12-26 11:41:33.921 - debug: mielecloudservice.1 (493619) Plugin sentry destroyed because not initialized correctly
2023-12-26 11:41:34.049 - info: mielecloudservice.1 (493619) starting. Version 6.5.4 in /opt/iobroker/node_modules/iobroker.mielecloudservice, node: v18.19.0, js-controller: 5.0.17
2023-12-26 11:41:34.074 - info: mielecloudservice.1 (493619) Login attempt #1 @Miele-API
2023-12-26 11:41:34.282 - silly: mielecloudservice.1 (493619) States system redis pmessage system.adapter.mielecloudservice.1.logLevel/system.adapter.mielecloudservice.1.logLevel:{"val":"silly","ack":true,"ts":1703587294248,"q":0,"from":"system.adapter.mielecloudservice.1","lc":1703586924845}
2023-12-26 11:41:34.357 - error: mielecloudservice.1 (493619) Error: Unable to authenticate user! Your credentials seem to be invalid. Please double check and fix them.
2023-12-26 11:41:34.359 - warn: mielecloudservice.1 (493619) Credentials used for login:
2023-12-26 11:41:34.359 - warn: mielecloudservice.1 (493619) options Miele_account: [xxx]
2023-12-26 11:41:34.359 - warn: mielecloudservice.1 (493619) options Miele_Password: [xxx]
2023-12-26 11:41:34.360 - warn: mielecloudservice.1 (493619) options Client_ID: [xxx]
2023-12-26 11:41:34.360 - warn: mielecloudservice.1 (493619) options Client_Secret: [xxx]
2023-12-26 11:41:34.360 - warn: mielecloudservice.1 (493619) options country: [nl-BE]
2023-12-26 11:41:34.361 - error: mielecloudservice.1 (493619) IMPORTANT!! Mask/Delete your credentials when posting your log online!
2023-12-26 11:41:34.361 - info: mielecloudservice.1 (493619) Login attempt wasn't successful. Connection retry in 31.134 Seconds.
2023-12-26 11:41:34.363 - info: mielecloudservice.1 (493619) Terminated (ADAPTER_REQUESTED_TERMINATION): Terminating adapter due to inability to authenticate.
2023-12-26 11:41:34.932 - info: host.rpi4 instance system.adapter.mielecloudservice.1 terminated by request of the instance itself and will not be restarted, before user restarts it.
```


CPU: 44,23 %
RAM: 25 %
Uptime: 0h
Available: 5.0.17
Installed: 5.0.17
Events: ?15 / ?25
Platform: linux
operating system: linux
Architecture: arm64
CPUs: 4
Speed: 1800 MHz
Model: Cortex-A72
RAM: 7.6 GB
System uptime: 14 d. 21:12:21
Node.js: v18.19.0
time: 1703586333894
timeOffset: -60
NPM: 10.2.5
adapters count: 615
Disk size: 234.2 GB
Disk free: 216.6 GB
Active instances: 4
Path: /opt/iobroker/
Uptime: 00:00:48
alive: true
_nodeCurrent: 18.19.0
_nodeNewest: 18.19.0
_nodeNewestNext: 18.19.0
_npmCurrent: 10.2.5
_npmNewest: 10.2.3
_npmNewestNext: 10.2.3

Thanks in advance and happy holidays!
